lib/src/tile_map_parser.dart

```diff
@@ -33,19 +34,47 @@ class TileMapParser {
       }
     });
 
+    map.properties = TileMapParser._parseProperties(
+        TileMapParser._getPropertyNodes(xmlElement));
+
     return map;
   }
 
   static Image _parseImage(XmlElement node) {
     var attrs = node.getAttribute;
-    return new Image(attrs('source'), int.parse(attrs('width')), int.parse(attrs('height')));
+    return new Image(
+        attrs('source'), int.parse(attrs('width')), int.parse(attrs('height')));
   }
 
-  static Map<String, String> _parseProperties(nodes) {
-    var map = new Map<String, String>();
-    nodes.forEach( (property) {
+  static Map<String, dynamic> _parseProperties(nodes) {
+    var map = <String, dynamic>{};
+
+    nodes.forEach((property) {
       var attrs = property.getAttribute;
-      map[attrs('name')] = attrs('value');
+      var value = attrs('value');
+      var name = attrs('name');
+
+      switch (attrs('type')) {
+        case 'bool':
+          map[name] = value == 'true' ? true : false;
+          break;
+        case 'int':
+          map[name] = int.parse(value);
+          break;
+        case 'float':
+          map[name] = double.parse(value);
+          break;
+        case 'color':
+          final a = value.substring(1, 3);
+          final rgb = value.substring(3);
+
+          map[name] = '#' + rgb + a;
```

Comment:
I'm not sure what this `color` type entails but on Flame/Flutter we often use colors as ARGB, not RGBA. (eg Color class constructor)

Reply:
Is there method to check if environment is web? And then optional parse to RGBA. In web ARGB is useless.

Reply:
hum I thought dart would standardize that... let's leave this as is but please add a todo comment, it's clearly an edge case and we can study this later. but I prefer to follow dart conventions, flutter-web should be translating whatever it needs.